docs/architecture/adr-018-network-upgrades.md

```diff
@@ -0,0 +1,68 @@
+# ADR 018: Network Upgrades
+
+## Status
+
+Proposed
+
+## Changelog
+
+- 2023/03/29: Initial draft
+- 2023/08/29: Modified to reflect the proposed decision and the detailed design
+
+## Context
+
+There are three requirements in Celestia's upgrading mechanism that diverges from how the standard Cosmos SDK currently operates:
+
+- The ability for non-validators to easily exit the system.
+- Minimal downtime. As a core piece of infrastructure that the rollup community depends on, Celestia needs to be highly available. This means reducing the downtime for upgrades through effective coordination and minimal migrations.
+- Full chain backwards compatibility. The latest software version should be able to correctly process all transactions of the chain since genesis.
```

Comment:
I think it meant to be
```
to correctly process all transactions of the chain since genesis in accordance with the version of the app that was in use at the time of each transaction's block.
```

Reply:
Yup I will extend this